main.tf

```diff
@@ -1,151 +1,213 @@
-  statement {
-    actions = ["sts:AssumeRole"]
-    resources = ["${aws_iam_role.admin.arn}"]
-  }
-}
-resource "aws_iam_policy" "assume_role_admin" {
-  name        = "AssumeRoleAdmin"
-  policy      = "${data.aws_iam_policy_document.assume_role_admin.json}"
-}
-
-data "aws_iam_policy_document" "assume_role_readonly" {
-  statement {
-    actions = ["sts:AssumeRole"]
-    resources = ["${aws_iam_role.readonly.arn}"]
   }
 }
-resource "aws_iam_policy" "assume_role_readonly" {
-  name        = "AssumeRoleReadOnly"
-  policy      = "${data.aws_iam_policy_document.assume_role_readonly.json}"
-}
 
 data "aws_iam_policy_document" "manage_mfa" {
   statement {
     sid = "AllowUsersToCreateEnableResyncDeleteTheirOwnVirtualMFADevice"
+
     actions = [
       "iam:CreateVirtualMFADevice",
       "iam:EnableMFADevice",
       "iam:ResyncMFADevice",
-      "iam:DeleteVirtualMFADevice"
+      "iam:DeleteVirtualMFADevice",
     ]
+
     resources = [
       "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/&{aws:username}",
-      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/&{aws:username}"
+      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/&{aws:username}",
     ]
   }
+
   statement {
     sid = "AllowUsersToDeactivateTheirOwnVirtualMFADevice"
+
     actions = [
-      "iam:DeactivateMFADevice"
+      "iam:DeactivateMFADevice",
     ]
+
     resources = [
       "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/&{aws:username}",
-      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/&{aws:username}"
+      "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/&{aws:username}",
     ]
+
     condition {
       test     = "Bool"
       variable = "aws:MultiFactorAuthPresent"
       values   = ["true"]
     }
   }
+
   statement {
     sid = "AllowUsersToListMFADevicesandUsersForConsole"
+
     actions = [
       "iam:ListMFADevices",
       "iam:ListVirtualMFADevices",
-      "iam:ListUsers"
+      "iam:ListUsers",
     ]
+
     resources = [
-      "*"
+      "*",
     ]
   }
 }
-resource "aws_iam_policy" "manage_mfa" {
-  name        = "AllowUsersToDeactivateTheirOwnVirtualMFADevice"
-  policy      = "${data.aws_iam_policy_document.manage_mfa.json}"
-}
 
 data "aws_iam_policy_document" "allow_change_password" {
   statement {
-    actions = ["iam:ChangePassword"]
+    actions   = ["iam:ChangePassword"]
     resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/&{aws:username}"]
   }
+
   statement {
-    actions = ["iam:GetAccountPasswordPolicy"]
+    actions   = ["iam:GetAccountPasswordPolicy"]
     resources = ["*"]
   }
 }
-resource "aws_iam_policy" "allow_change_password" {
-  name        = "AllowChangePassword"
+
+# Admin config
+
+resource "aws_iam_policy" "manage_mfa_admin" {
+  name        = "${module.admin_label.id}${var.delimiter}AllowUsersToDeactivateTheirOwnVirtualMFADevice"
```

Comment:
not keen on mixing camel case with hyphens. Also, we don't need the precision of `AllowUsersToDeactivateTheirOwnVirtualMFADevice`. 

e.g. 

cp-prod-mfa-management is good enough.